programming/Encoders.md

```diff
@@ -0,0 +1,53 @@
+
+There's lots of different examples of encoders at work all around us, 
+such as:
+  * Elevators - Encoders in elevators tell the elevator's controller device its position so that it can properly align itself with the floor and open up the doors at the correct position.
+  * Cranes - Some large cranes use encoders in order to collect accurate location information to know where it is.
+
+**TL;DR:  Encoders are the reason we can tell our robot to drive at angles.**
+
+**How do Encoders (generally) work?**
+
+<img src="http://encoder.com/core/files/encoder/uploads/images/Encoder-exploded-COLOR-v2.jpg"></img>
+
+There are a lot of encoders in the world at large.  Hardware encoders are made out of a variety of different technologies and tend to vary quite a bit as do their purposes.  The encoder in the example shown above is used to detect the speed of which the motor is spinning.  This specific encoder, taken from <a href="http://encoder.com/blog/company-news/what-is-an-encoder/">encoder.com</a>, has a stream of light which comes from the LED and goes through the "Code Disk".  As the light goes through the "Code Disk" it becomes interrupted as the blades from the "Code Disk" block it from the "Photodetector Assembly" which detects light.  The result is a binary output of either the light being on or the light being off.  This simple value is then passed on to another device, generally called a "controller" which actually interprets the data.  Because the controller can obtain these simple values it can measure how frequently the light is being blocked from which it can determine the speed of the motor.  It can check whether or not the motor is actually running based on if there is no change in the on or off value over a long period of time.
+
+Most optical sensors work like the one pictured above by relying on the interruption of light picked up by another sensor.
+
+## How we use encoders!
+
+We use the *robotpy library* for interfacing to our robot's components with Python, and the library has a class interface to <a href="http://robotpy.readthedocs.io/projects/wpilib/en/latest/wpilib/Encoder.html">the Encoder library</a> (yes they call it the encoder library), and to use it, you first create an instance of Encoder oriented to your hardware encoder, such as: **[EXAMPLES OF ENCODERS IN THE ROBOTPY LIBRARY TO BE ADDED]**
+```python
+# Here is an example by Ben from the programming team of the library (code untested)
+
+def main():
+
+     channel = 0
+     pot = AnalogPotentiometer(channel)
+     
+     print(pot.get())
```

Comment:
Seems like we would want a little bit more in this example than simply printing it, and if we print it, what are the typical values? How is the value used in a typical scenario?